p256/src/arithmetic/hash2curve.rs

```diff
@@ -0,0 +1,324 @@
+use elliptic_curve::bigint::{ArrayEncoding, U256};
+use elliptic_curve::consts::U48;
+use elliptic_curve::generic_array::GenericArray;
+use elliptic_curve::group::cofactor::CofactorGroup;
+use elliptic_curve::hash2curve::OsswuMap;
+use elliptic_curve::subtle::{Choice, CtOption};
+use elliptic_curve::{
+    hash2curve::{GroupDigest, MapToCurve, OsswuMapParams, Sgn0},
+    hash2field::FromOkm,
+};
+
+use crate::{AffinePoint, NistP256, ProjectivePoint, Scalar};
+
+use super::FieldElement;
+
+impl GroupDigest for NistP256 {
+    type FieldElement = FieldElement;
+    type Output = ProjectivePoint;
+}
+
+impl FromOkm for FieldElement {
+    type Length = U48;
+
+    fn from_okm(data: &GenericArray<u8, Self::Length>) -> Self {
+        const F_2_192: FieldElement = FieldElement([
+            0xffff_fffe_ffff_ffffu64,
+            0xffff_ffff_ffff_fffeu64,
+            0x0000_0002_0000_0000u64,
+            0x0000_0000_0000_0003u64,
+        ]);
+        let d0 = FieldElement([
+            u64::from_be_bytes(data[16..24].try_into().unwrap()),
+            u64::from_be_bytes(data[8..16].try_into().unwrap()),
+            u64::from_be_bytes(data[0..8].try_into().unwrap()),
+            0,
+        ])
+        .to_montgomery();
+        let d1 = FieldElement([
+            u64::from_be_bytes(data[40..48].try_into().unwrap()),
+            u64::from_be_bytes(data[32..40].try_into().unwrap()),
+            u64::from_be_bytes(data[24..32].try_into().unwrap()),
+            0,
+        ])
+        .to_montgomery();
+        d0 * F_2_192 + d1
+    }
+}
+
+impl Sgn0 for FieldElement {
+    fn sgn0(&self) -> Choice {
+        FieldElement::sgn0(self)
```

Comment:
Seems like you could just call `self.is_odd()` here rather than adding a separate inherent method in addition to the trait method

Reply:
Done.